scripts/jenkins/ardana/gerrit/gerrit.py

```diff
@@ -0,0 +1,235 @@
+        given branch
+        """
+        query = '/changes/?q={}'.format(change_id)
+        if branch:
+            query += '+branch:{}'.format(branch)
+        if patchset:
+            query += '&o=ALL_REVISIONS&o=ALL_COMMITS&o=DETAILED_LABELS'
+        else:
+            query += '&o=CURRENT_REVISION&o=CURRENT_COMMIT&o=DETAILED_LABELS'
+        response_json = self._query_gerrit(query)
+
+        if len(response_json) > 1:
+            raise Exception(
+                "Unable to get a unique change for {}{}."
+                "This can also happen if the same change-id is used "
+                "in multiple gerrit projects.".format(
+                    change_id,
+                    " and branch {}".format(branch) if branch else '')
+            )
+        elif len(response_json) != 1:
+            raise Exception("Unable to find a change for {}{}".format(
+                change_id, " and branch {}".format(branch) if branch else ''))
+
+        self._change_object = response_json[0]
+
+    def _set_change_object(self, change_object):
+        self._change_object = change_object
+
+    def _load_change_object(self, patchset=None):
+        self.id = str(self._change_object['_number'])
+        self.change_id = self._change_object['change_id']
+        self.gerrit_project = self._change_object['project'].split('/')[1]
+        self.status = self._change_object['status']
+        if not patchset:
+            current_revision = self._change_object['current_revision']
+            revision_obj = self._change_object['revisions'][current_revision]
+        else:
+            revisions = [r for r in self._change_object['revisions'].values()
+                         if r['_number'] == int(patchset)]
+            if len(revisions) != 1:
+                print json.dumps(self._change_object, indent=4)
+                raise Exception(
+                    "Unable to find patchset {} for change {} ".format(
+                        patchset, self.id))
+            revision_obj = revisions[0]
+        self.patchset = str(revision_obj['_number'])
+        self.url = revision_obj['fetch']['anonymous http']['url']
+        self.ref = revision_obj['fetch']['anonymous http']['ref']
+        self.branch = self._change_object['branch']
+        self.subject = revision_obj['commit']['subject']
+        self.commit_message = revision_obj['commit']['message']
+        self.mergeable = self._change_object.get('mergeable', False)
+        self.submittable = self._change_object.get('submittable', False)
+        self.labels = self._change_object['labels']
+
+    def load_dependencies(self, depth=None, loaded_deps=[]):
+        if depth == 0:
+            return
+        elif depth is not None:
+            depth -= 1
+        deps = find_dependency_headers(self.commit_message)
+        for change_id in deps:
+            change = GerritChange(change_id, self.branch)
+            if change.id in loaded_deps:
+                continue
```

Comment:
This will not catch circular dependencies. It's okay if two items depend on the same change, but if one of those children depend on a parent that could cause issues. We probably need to track the loaded deps as a acyclic graph.

Reply:
For the purposes of this function, we don't need to catch circular dependencies, we only need to avoid them. What this method does is it reads the Depends-On graph and transforms it into a tree. I do agree the implementation is weird though, as it doesn't really make sense to save a list of dependencies for every change object. I will change the implementation to return a list of GerritChange objects representing the complete list of dependencies, without storing them internally in the GerritChange object, because the tree structure is not relevant anywhere else in the code.

Reply:
I simplified this method. The docstring should explain what it does.

Reply:
I can't see this change, am I missing something or was it not pushed?